File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    const stack = [];
    const mySet = new Set(['+','-','/','*']);
    for(let i=0;i<tokens.length;i++){
        const token = tokens[i];
        if(mySet.has(token)){
            const a = Number(stack.pop());
            const b=Number(stack.pop());
            let result = 0;
            if(token == '+'){
                result = a+b;

            }
            else if(token == '-'){
                result = b-a;
            }
            else if(token =='*'){
                result = b*a;
            }else if(token === '/'){
              
                result = Math.trunc(b/a);
                
            }
            stack.push(result);
            
        

        }else{
            stack.push(token);
      
        }
    }
    return Number(stack.pop());
    
};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) { 
        Stack<Integer> stack = new Stack<>();
        HashSet<String> hset = new HashSet<>();
        hset.add("+");
        hset.add("-");
        hset.add("*");
        hset.add("/");
        for(int i=0;i<tokens.length;i++){
            String  s = tokens[i];
            if(hset.contains(s)){
                int a = stack.pop();
                int b = stack.pop();
                int result = 0;
                if(s.equals("+")){
                     result = b+a;
                }else if(s.equals("-")){
                      result  = b-a;
                }else if(s.equals("*")){
                    result = b*a;
                }else{
                    result =( b/a);
                }
                stack.push(result);
               
            }else{
                stack.push(Integer.parseInt(s));
            }
        }

        return stack.pop();
        
        
    }
}